pgjdbc/src/main/java/org/postgresql/core/v3/ConnectionFactoryImpl.java

```diff
@@ -687,12 +703,25 @@ private static void doAuthentication(PGStream pgStream, String host, String user
 
     /* SCRAM authentication state, if used */
     ScramAuthenticator scramAuthenticator = null;
+    // TODO: figure out how to deal with new protocols
+    int protocol = 3 << 16;
 
     try {
       authloop: while (true) {
         int beresp = pgStream.receiveChar();
 
         switch (beresp) {
+          case 'v':  // Negotiate Protocol Version
+            // read the length and ignore it.
+            pgStream.receiveInteger4();
+            protocol = pgStream.receiveInteger4();
+            int numOptionsNotRecognized = pgStream.receiveInteger4();
+            String [] optionsNotRecognized = new String[numOptionsNotRecognized];
+            for (int i = 0; i < numOptionsNotRecognized; i++) {
+              optionsNotRecognized[i] = pgStream.receiveString();
+            }
+            pgStream.setProtocolVersion(new ProtocolVersion(protocol));
```

Comment:
Did you intend something as follows here?

```suggestion
            pgStream.setProtocolVersion(ProtocolVersion.fromMajorMinor(...protocol));
```

Reply:
We probably should throw an exception here as the connection should not be completed with invalid options